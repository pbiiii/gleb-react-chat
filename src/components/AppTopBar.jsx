import React from 'react';
import {withStyles} from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Typography from "@material-ui/core/es/Typography/Typography";
import Toolbar from '@material-ui/core/Toolbar';
import { UserMenu, ChatMenu } from "@src/components";

const drawerWidth = 320;

const styles = theme => ({
    appBar: {
        position: 'fixed',
        width: 'calc(100% - 320px)',
        marginLeft: drawerWidth,
    },
    appBarTitle: {
        display: 'flex',
        justifyContent: 'space-between',
    },
});

const AppBarComponent = (
    {
        classes,
        activeChat = {title: 'Active chat'},
        isConnected,
        activeUser,
        leaveChat,
        deleteChat,
        logout,
        editUser
    }
) => {
    return (
        <AppBar position="fixed" className={classes.appBar}>
            <Toolbar className={classes.appBarTitle}>
                {activeChat ? (
                    <ChatMenu
                        disabled={!isConnected}
                        activeUser={activeUser}
                        onLeaveClick={() => leaveChat(activeChat._id)}
                        activeChat={activeChat}
                    />
                ) : (
                    <Typography variant="title" >
                        Gleb React Chat
                    </Typography>
                )}
                <UserMenu
                    disabled={!isConnected}
                    activeUser={activeUser}
                    onLogoutClick={logout}
                    onEditProfileClick={editUser}
                />
            </Toolbar>
        </AppBar>
    );
}

export const AppTopBar = withStyles(styles)(AppBarComponent);