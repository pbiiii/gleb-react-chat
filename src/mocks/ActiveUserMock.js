import UserMock from './UserMock';

export default {
    ...UserMock,
    isMember: false,
    isCreator: true,
    isChatMember: true,
};
