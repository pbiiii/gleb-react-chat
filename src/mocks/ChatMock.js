import UserMock from './UserMock';

export default {
    _id: 'ashasdaksdmkl1mkl',
    creator: { ...UserMock, _id: 2 },
    title: 'Mock title',
};
