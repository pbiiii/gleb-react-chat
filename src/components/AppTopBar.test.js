/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import AppTopBar from './AppTopBar';
import ActiveUserMock from '../mocks/ActiveUserMock';
import ChatMock from '../mocks/ChatMock';

jest.mock('./UserMenu', () => () => 'UserMenu');
jest.mock('./ChatMenu', () => () => 'ChatMenu');

const mockProps = {
    activeUser: { ...ActiveUserMock },
    activeChat: { ...ChatMock },
    isConnected: true,
    deleteChat: () => {},
    leaveChat: () => {},
    logout: () => {},
    editUser: () => {},
};

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<AppTopBar {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('render correctly', () => {
    const tree = renderer
        .create(<AppTopBar {...mockProps} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
