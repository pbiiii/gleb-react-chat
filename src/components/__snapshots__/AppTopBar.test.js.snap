// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<header
  className="MuiPaper-root-54 MuiPaper-elevation4-60 MuiAppBar-root-45 MuiAppBar-positionFixed-46 MuiAppBar-colorPrimary-52 mui-fixed AppTopBar-appBar-43"
>
  <div
    className="MuiToolbar-root-81 MuiToolbar-regular-83 MuiToolbar-gutters-82 AppTopBar-appBarTitle-44"
  >
    ChatMenu
    UserMenu
  </div>
</header>
`;
